.github/pull_request_template.md

```diff
@@ -21,6 +21,17 @@ Describe the tests that you ran to verify your changes.
 
 Make sure to provide instructions for the maintainer as well as any relevant configurations.
 
+- [ ] Basic tests (linting, compilation, formatting, unit/integration tests)
```

Comment:
Won't the CI do these for you anyways? Do we need a checkbox for that?

Reply:
Yeah that's kind of the point of them :)

Reply:
@muXxer @DaughterOfMars The idea is to use the CI only as the last defense, relieve the runners and avoid preliminary reviews and efforts

Reply:
Are we gating the CI for external users? Or planning to?

Reply:
Not that I know of @DaughterOfMars

Reply:
Then it does seem redundant to also have this checkbox, as we know this is true if the CI passed

Reply:
@DaughterOfMars  I see what you mean but this suggests it's fine that authors rely on CI tests without running relevant tests locally.

Reply:
Well, not really. They will do that anyway unless we restrict it

Reply:
I see your point. My point is that we want this conditional redundancy for the reasons laid out above. Plus the CI scope extends beyond what is described with this clause.

This is not something new: We do have the `I have checked that new and existing unit tests pass locally with my changes` below as well. These are directions aimed at ensuring some minimal assurance guarantees before creating relying on the CI.

Reply:
I share the view. Running **_relevant_** unit / integration tests before putting the PR under review, including lints and format sounds reasonable - as said, it will avoid unnecessary review iterations. Locally you might not catch everything every time but that is fine if the CI then will catch it.

Reply:
Doesn't the `## Change checklist` already include all these things? Especially the last two points
```
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes
```
and if something specific is missing that you want to mention, then put it in the list and don't create a new list?

Reply:
Not, really I think. The additions reflect test classes which fit the test section. The checks you mentioned quoted I think don't really reflect that and I think we shouldn't move them to this more general checklist section.